Assets/MixedRealityToolkit/Definitions/Utilities/ArticulatedHandPose.cs

```diff
@@ -0,0 +1,286 @@
+                if (handedness == Handedness.Left)
+                {
+                    p.x = -p.x;
+                    r.y = -r.y;
+                    r.z = -r.z;
+                }
+
+                // Apply camera transform
+                p = cameraRotation * p;
+                r = cameraRotation * r;
+
+                // Apply external transform
+                p = position + rotation * p;
+                r = rotation * r;
+
+                jointsOut[i] = new MixedRealityPose(p, r);
+            }
+        }
+
+        public void ParseFromJointPoses(MixedRealityPose[] joints, Handedness handedness, Quaternion rotation, Vector3 position)
+        {
+            var invRotation = Quaternion.Inverse(rotation);
+            var invCameraRotation = Quaternion.Inverse(CameraCache.Main.transform.rotation);
+
+            for (int i = 0; i < jointCount; i++)
+            {
+                Vector3 p = joints[i].Position;
+                Quaternion r = joints[i].Rotation;
+
+                // Apply inverse external transform
+                p = invRotation * (p - position);
+                r = invRotation * r;
+
+                // To camera space
+                p = invCameraRotation * p;
+                r = invCameraRotation * r;
+
+                // Pose offset are for right hand, mirror on X axis if left hand is given
+                if (handedness == Handedness.Left)
+                {
+                    p.x = -p.x;
+                    r.y = -r.y;
+                    r.z = -r.z;
+                }
+
+                localJointPoses[i] = new MixedRealityPose(p, r);
+            }
+        }
+
+        public void SetZero()
+        {
+            for (int i = 0; i < jointCount; i++)
+            {
+                localJointPoses[i] = MixedRealityPose.ZeroIdentity;
+            }
+        }
+
+        public void Copy(ArticulatedHandPose other)
+        {
+            Array.Copy(other.localJointPoses, localJointPoses, jointCount);
+        }
+
+        public void InterpolateOffsets(ArticulatedHandPose poseA, ArticulatedHandPose poseB, float value)
+        {
+            for (int i = 0; i < jointCount; i++)
+            {
+                var p = Vector3.Lerp(poseA.localJointPoses[i].Position, poseB.localJointPoses[i].Position, value);
+                var r = Quaternion.Slerp(poseA.localJointPoses[i].Rotation, poseB.localJointPoses[i].Rotation, value);
+                localJointPoses[i] = new MixedRealityPose(p, r);
+            }
+        }
+
+        public void TransitionTo(ArticulatedHandPose other, float lastAnim, float currentAnim)
+        {
+            if (currentAnim <= lastAnim)
+            {
+                return;
+            }
+
+            float range = Mathf.Clamp01(1.0f - lastAnim);
+            float lerpFactor = range > 0.0f ? (currentAnim - lastAnim) / range : 1.0f;
+            for (int i = 0; i < jointCount; i++)
```

Comment:
>for (int i = 0; i < jointCount; i++) [](start = 12, length = 36)

Can we call InterpolateOffsets() instead?

Reply:
I'm going to remove this function. It's quite specific to how hand simulation animation works and doesn't make much sense as a generic function (it's just computing lerp factor based on previous value). Saves me the hassle of trying to explain what it does ;)